src/main/java/TaskList.java

```diff
@@ -0,0 +1,94 @@
+import java.util.regex.Pattern;
+import java.util.regex.Matcher;
+public class TaskList {
+    private static Task[] allTasks = {}; //array of inputs
+
+    public static void addToTaskList(String input, Task.TaskType taskType){
+        Task[] newTaskList = new Task[allTasks.length+1];
+        System.arraycopy(allTasks, 0, newTaskList, 0, allTasks.length);
+        switch (taskType){
+            case TODO:
+                String toDoDescription = input.split("todo")[1].strip();
+                newTaskList[allTasks.length] = new ToDo(toDoDescription);
+                break;
+            case DEADLINE:
+                String[] deadlineDetails = input.split("deadline")[1].strip().split("/");
+                String deadlineDescription = deadlineDetails[0].strip();
+                String deadlineTiming = deadlineDetails[1].strip().split("by")[1].strip();;
+                newTaskList[allTasks.length] = new Deadline(deadlineDescription, deadlineTiming);
+                break;
+            case EVENT:
+                String[] eventDetails = input.split("event")[1].strip().split("/");
+                String eventDescription = eventDetails[0].strip();
+                String eventStartTiming = eventDetails[1].strip().split("from")[1].strip();
+                String eventEndTiming = eventDetails[2].strip().split("to")[1].strip();;
+                newTaskList[allTasks.length] = new Event(eventDescription, eventStartTiming, eventEndTiming);
+                break;
+        }
```

Comment:
Try to follow the standard not to indent case clause. Similar issues in your other switch case statements.
```suggestion
        switch (taskType){
        case TODO:
            String toDoDescription = input.split("todo")[1].strip();
            newTaskList[allTasks.length] = new ToDo(toDoDescription);
            break;
        case DEADLINE:
            String[] deadlineDetails = input.split("deadline")[1].strip().split("/");
            String deadlineDescription = deadlineDetails[0].strip();
            String deadlineTiming = deadlineDetails[1].strip().split("by")[1].strip();;
            newTaskList[allTasks.length] = new Deadline(deadlineDescription, deadlineTiming);
            break;
        case EVENT:
            String[] eventDetails = input.split("event")[1].strip().split("/");
            String eventDescription = eventDetails[0].strip();
            String eventStartTiming = eventDetails[1].strip().split("from")[1].strip();
            String eventEndTiming = eventDetails[2].strip().split("to")[1].strip();;
            newTaskList[allTasks.length] = new Event(eventDescription, eventStartTiming, eventEndTiming);
            break;
        }
```